### DISCUSSION TITLE
How to handle passwords in talkers

### DISCUSSION THREAD

**Discussion by mizaki** [comictagger/comictagger#464] [Ideas]
I have a working Metron talker (limited API return data not withstanding) but it requires a username and password to access the API (or derived HTTP Basic key from said). Currently the password is stored as text and the GUI generates a normal text field.

I don't see much that can be done about saving the password as plain text in the config file. It's also not all that much of a worry for people to steal it etc. imo.

When it comes to the GUI, I could add a check to see if the field name ends with "password" and then use a QT password field.

I'm wondering if you have any thoughts or suggestions on any of that?

**Comment by lordwelch** [comment]
Password=API key
If anything an API key is more dangerous for a particular service because people think it's safer. It's only safer because people are giving out the password that they re-used for every other service to less applications.

**Reply by mizaki** [reply]
If I understand correctly: Put the password in the API key field and also have that field be a password in the GUI so it's hidden (with show button)?

**Reply by lordwelch** [reply]
If you want to make it a password field with a show button you can but it doesn't need to be, anyone that edits the config via the file will just see the plaintext password
